framework/python/src/common/session.py

```diff
@@ -528,16 +528,14 @@ def update_profile(self, profile_json):
     return risk_profile
 
   def check_profile_status(self, profile):
+    """Check if a profile has expired"""
 
-    if profile.status == 'Valid':
+    created_date = profile.created.timestamp()
 
-      # Check expiry
-      created_date = profile.created.timestamp()
+    today = datetime.datetime.now().timestamp()
 
-      today = datetime.datetime.now().timestamp()
-
-      if created_date < (today - SECONDS_IN_YEAR):
-        profile.status = 'Expired'
+    if created_date < (today - SECONDS_IN_YEAR):
+      profile.status = 'Expired'
 
     return profile.status

```

Comment:
This should be a direct call to the profile where expiration status is already determined.  Since that is a private method, likely a new method should be created there for updating the status only,i.e. 
```

profile.update_status()
return profile.status
```